connector/connect/src/main/protobuf/spark/connect/base.proto

```diff
@@ -48,6 +54,9 @@ message Request {
   // The logical plan to be executed / analyzed.
   Plan plan = 3;
 
+  // From which client this request was sent from.
+  ClientType client_type = 4;
```

Comment:
To avoid changing this all the time a new client comes along, this should most likely be a string.
```suggestion
  optional string client_type = 4;
```